app/admin-dashboard/page.tsx

```diff
@@ -1,5 +1,32 @@
+'use client';
+
+import React, { useState } from 'react';
+import ReusableModal from '../../components/reusable-components/reusable-modal/ReusableModal';
+
 const AdminDashboard = () => {
-  return <div>Admin Dashboard</div>;
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const showModal = () => {
+    setIsModalOpen(true);
+  };
+
+  const closeModal = () => {
+    setIsModalOpen(false);
+  };
+
```

Comment:
since we will be using the modal in multiple components, we would need to write this logic in all of them.
Lets try to make this more reusable.
(hint: use context)